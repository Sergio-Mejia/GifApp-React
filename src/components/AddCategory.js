import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');
    
    const handelInputChange = (e) =>{
        setinputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length > 2){ //Trim para eliminar los espacios
            setCategories(cats => [inputValue, ...cats]); //agregar elemento al arreglo
            setinputValue(''); //dejar la caja de texto vac√≠a 
        }

        // console.log('submit hecho');
        
    }

    return (

        <form action="" onSubmit={handleSubmit}>
            <input
                type='text'
                value={ inputValue }
                onChange={handelInputChange}
            />
        </form>
    );
};


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
