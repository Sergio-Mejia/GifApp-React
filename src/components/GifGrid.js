import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifgridItem';

export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);
            //data:images -> renombrar data como images

    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>} {/*operador && funciona como if sin evaluar el else */}
            <div className='card-grid'>
                    {
                        images.map( image => (
                            <GifGridItem 
                                key = {image.id}
                                {...image}  //mandar cada una de las propiedades de la imagen
                            />
                        ))
                    }
            </div>
        </>
    )
}