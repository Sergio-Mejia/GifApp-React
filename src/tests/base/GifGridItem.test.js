import React from 'react';
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifgridItem';



describe('pruebas en GifGridItem', () => { 

    const title = 'Un titulo';
    const  url = 'https://localhost/some.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Debe mostrar el componente correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);

    });


    test('debe tener la imagen igual a la url y alt de los props', () => { 

        const img = wrapper.find('img');
        // console.log(img.props('src'));
        expect(img.prop('src')).toBe(url);
        expect(img.props().alt).toBe(title);
    })

    
    test('debe tener animate__FadeInUp', () => { 

        const div = wrapper.find('div');
        const className = div.props().className;

        expect(className.includes('animate__fadeInUp')).toBe(true);
    })
})