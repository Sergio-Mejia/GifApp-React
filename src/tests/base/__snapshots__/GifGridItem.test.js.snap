// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Un titulo"
    src="https://localhost/some.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
